### ISSUE TITLE
Processing RethinkDB returns undeterministic output

### ISSUE THREAD

**Issue by robert-zaremba** [dancannon/gorethink#64]
I'm not sure if this is a RethinkDB bug or gorethinkdb one, but Python driver and JavaScript one behaves normally.

How to replicate:
- insert a lot of values into a table
- try to sum them in an application side
- run several times - you will observe different results.

I wrote a scripts (https://gist.github.com/robert-zaremba/8411925) which show the issue. Run:
- `python values.py yes`   (yes to create missing table and insert variables)
- `go run values.go`  - it should return different result (run it several times)

**Comment by dancannon** [comment]
I just tried running your example code and I got the same value every time and this value is the same as the value I received when running a reduce query in the web admin panel.

The only thing I can think that would cause this is that you might have been running the go program before the writes were written to the disk or I have change something before which fixed this issue.

**Comment by robert-zaremba** [comment]
I forget to add `run()` to `table_create`. I've also added some debug prints to a Python table creation/feed code.

I tested on a different node and in Go it returned the same values (maybe on my localhost I have some memory issue). However the Go results are still different from those from Python, and DataExplorer reduction function:

```
r.db('test').table('values').map(r.row('v')).reduce(function(acc, val) { return acc.add(val)}, 0)
```

When inserting values `durability('soft')` is enough because we don't need to have that durability (we will know if DB is down). Anyway multiple python runs returns the same value as the JS reduction function, and Go is a bit different. I will ask others to test this issue.
